src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -34,46 +37,52 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // Boolean to update User's main page 
     Boolean showNewTab = false;
 
+    User user;
     if (userService.isUserLoggedIn()) {
-      userEmail = userService.getCurrentUser().getEmail();
       userId = userService.getCurrentUser().getUserId();
+      userEmail = userService.getCurrentUser().getEmail();
       logoutUrl = userService.createLogoutURL(urlToRedirect);
       showNewTab = true;
+      user = new User(userId, userEmail);
     } else {
       loginUrl = userService.createLoginURL(urlToRedirect);
+      user = new User(userId, userEmail);
```

Comment:
You can probably put "user = new User(userId, userEmail);" outside of the if statement because the statement is used either way, but honestly it's up to you if you feel like doing that or not.

Reply:
The value of userId changes based on whether or not the user is logged in.

Reply:
I see what y'all mean!

Reply:
Did this result in a change? Looks like user = new User is still repeated in the if/else clause... Seems like it could go right after the clause...

Reply:
Fixed!